edg_acoustics/time_integration.py

```diff
@@ -39,16 +40,69 @@ def step_dt(self):
         pass
 
 class TSI_TI(TimeIntegrator):
-    def __init__(self, AcousticSimulation_object, CFL: float=CFL_Default):
+    def __init__(self, AcousticSimulation_object, L_operator: typing.Callable[[numpy.array, numpy.array, numpy.array, numpy.array]], Nt: int, CFL: float=CFL_Default):
         # self.BC = BC_object
         self.sim = AcousticSimulation_object
+        self.L_operator = L_operator  # the function in AcousticSimulation that enables the computation of Lq, given q = [P, Vx, Vy, Vz]
+        self.Nt = Nt  # degree of time integration
         self.CFL = CFL
         self.dt = CFL * self.sim.dtscale / self.sim.c0 / (2 * self.sim.Nx +1)
         self.dVx = numpy.zeros_like(AcousticSimulation_object.Fscale)
         self.dVy = numpy.zeros_like(self.dVx)
         self.dVz = numpy.zeros_like(self.dVx)
         self.dP = numpy.zeros_like(self.dVx)
-        
+
+    def step_dt_new(self, P0, Vx0, Vy0, Vz0):
+         # Takes the pressure, velocity at the time T and evolves it to time T + dt
+    #   P0 := P(T), P0 contains the (high-order) derivative values as well 
+    #   P := P(T + dt)
+    # the same for all the other variables
+##########################
+
+        P = P0.copy()
+        Vx = Vx0.copy()
+        Vy = Vy0.copy()
+        Vz = Vz0.copy()
+        # print(f"inside, P ID {id(P)}, sim.P ID {id(self.sim.P)}")
+
+        for Tind in range (1, self.Nt+1):
+            # Compute L (L^{Tind-1} q)
+            P0, Vx0, Vy0, Vz0 = self.L_operator(P0, Vx0, Vy0, Vz0)
+
+            # Add the Taylor term \frac{dt^{Tind}}{Tind!}L^{Tind}q
+            Vx += self.dt**Tind / math.factorial(Tind) * Vx0
+            Vy += self.dt**Tind / math.factorial(Tind) * Vy0
+            Vz += self.dt**Tind / math.factorial(Tind) * Vz0
+            P += self.dt**Tind / math.factorial(Tind) * P0
+
+            # We need to look into this, I think this can also be included in self.sim.L
+            # Or better even, added as another function to treat the boundary conditions
```

Comment:
These need to be modified carefully. I can do it.

Reply:
Good, but if, as you proposed, we pass BC to `step_dt_new` then these lines are ok, no?